compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/test/fuse_and_hoist_forall.mlir

```diff
@@ -304,3 +304,39 @@ func.func @multi_hoist_with_other_ops_in_loop(%2: tensor<128x128xf16>, %3: tenso
 //       CHECK:     scf.forall.in_parallel
 //       CHECK:   scf.forall.in_parallel
 //       CHECK:   return
+
+// -----
+
+func.func @hoist_with_single_trip_loops(%2: tensor<128x128xf16>, %3: tensor<128x128xf16>) -> tensor<128x128xf16> {
+  %c4 = arith.constant 4 : index
+  %c128 = arith.constant 128 : index
+  %c0 = arith.constant 0 : index
+  %empty = tensor.empty() : tensor<128x128xf16>
+  %8 = scf.for %arg0 = %c0 to %c128 step %c4 iter_args(%arg1 = %empty) -> (tensor<128x128xf16>) {
+    %9 = scf.forall (%arg2, %arg3) in (1, 1) shared_outs(%arg4 = %arg1) -> (tensor<128x128xf16>) {
```

Comment:
Wouldnt it just be easier to remove this `scf.forall` then you dont need a special pattern/pass for it.

Reply:
This comment seems like it's for #18418, but we cannot fold away this `scf.forall` unless the static workgroup size is exactly `1 subgroup`, otherwise when resolving this `scf.forall` it needs to become an `scf.if` (or `scf.for`). In this particular case we could fold away the loop, but since we can't fold away such loops in general anyway, I first extended the hoisting pattern.

Reply:
Since this comment was for a different commit I'm going to land this part of the change.